src/lib/systemd.ts

```diff
@@ -1,113 +1,130 @@
 
-const bus = dbus.systemBus();
-const invokeAsync = Bluebird.promisify(bus.invoke).bind(bus);
+export class DbusError extends TypedError {}
+
+const bus = dbus.getBus('system');
+const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
+	context: bus,
+});
 
 const defaultPathInfo = {
 	path: '/org/freedesktop/systemd1',
 	destination: 'org.freedesktop.systemd1',
 	interface: 'org.freedesktop.systemd1.Manager',
 };
 
-const systemdManagerMethodCall = (
-	method: string,
-	signature?: string,
-	body?: any[],
+async function getInterface(
 	pathInfo: {
 		path: string;
 		destination: string;
 		interface: string;
 	} = defaultPathInfo,
-) => {
-	if (signature == null) {
-		signature = '';
-	}
-	if (body == null) {
-		body = [];
+) {
+	try {
+		return await getInterfaceAsync(
+			pathInfo.destination,
+			pathInfo.path,
+			pathInfo.interface,
+		);
+	} catch (e) {
+		throw new DbusError(e);
 	}
-	return invokeAsync({
-		...pathInfo,
-		member: method,
-		signature,
-		body,
-	});
-};
+}
 
-export function restartService(serviceName: string) {
-	return systemdManagerMethodCall('RestartUnit', 'ss', [
-		`${serviceName}.service`,
-		'fail',
-	]);
+export async function restartService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
+		systemd.RestartUnit(`${serviceName}.service`, 'fail');
+	} catch (e) {
+		throw new DbusError(e);
+	}
 }
 
-export function startService(serviceName: string) {
-	return systemdManagerMethodCall('StartUnit', 'ss', [
-		`${serviceName}.service`,
-		'fail',
-	]);
+export async function startService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
+		systemd.StartUnit(`${serviceName}.service`, 'fail');
+	} catch (e) {
+		throw new DbusError(e);
+	}
 }
 
-export function stopService(serviceName: string) {
-	return systemdManagerMethodCall('StopUnit', 'ss', [
-		`${serviceName}.service`,
-		'fail',
-	]);
+export async function stopService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
+		systemd.StopUnit(`${serviceName}.service`, 'fail');
+	} catch (e) {
+		throw new DbusError(e);
+	}
 }
 
-export function enableService(serviceName: string) {
-	return systemdManagerMethodCall('EnableUnitFiles', 'asbb', [
-		[`${serviceName}.service`],
-		false,
-		false,
-	]);
+export async function enableService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
+		systemd.EnableUnitFiles([`${serviceName}.service`], false, false);
+	} catch (e) {
+		throw new DbusError(e);
+	}
 }
 
-export function disableService(serviceName: string) {
-	return systemdManagerMethodCall('DisableUnitFiles', 'asb', [
-		[`${serviceName}.service`],
-		false,
-	]);
+export async function disableService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
+		systemd.DisableUnitFiles([`${serviceName}.service`], false);
+	} catch (e) {
+		throw new DbusError(e);
+	}
 }
 
 export const reboot = Bluebird.method(() =>
```

Comment:
Why is `Bluebird.method`needed here ?
Also this returns undefined and the caller seems to be awaiting on it.
This could be 
```typescript
await delay(1000);
const systemd = await ...
systemd.Reboot(false);
```
and the await in the caller can be removed.